src/main/java/ui/TestController.java

```diff
@@ -157,4 +159,30 @@ public static RepoIO createTestingRepoIO(Optional<JSONStore> jsonStoreToBeUsed)
         return new RepoIO(Optional.of(new DummySource()), jsonStoreToBeUsed,
                           Optional.of(RepoStore.TEST_DIRECTORY));
     }
+
+    /**
+     * Create an update progress window if not in test mode.
+     *
+     * @return update progress window
+     */
+    public static UpdateProgressWindow createUpdateProgressWindow() {
+        if (isTestMode()) {
+            return null;
```

Comment:
The test class in problem is `PlatformExTest` because it does not set the test mode to true.